xuluomin/server/src/sever_node.cpp

```diff
@@ -0,0 +1,58 @@
+#include <ros/ros.h>
+#include <client/Time.h>
+#include <client/show.h>
+#include "std_msgs/Header.h"
+
+bool show_data(const std_msgs::Header::ConstPtr& msg) {
+    // 打印接收到的数据
+    ROS_INFO("data name:%s,time:%d",
+             msg->frame_id.c_str(), msg->stamp.sec);
+}
+bool callback(client::show::Request& req,
+              client::show::Response& res) {
+    ros::NodeHandle n;
+    // 显示请求数据
+    ROS_INFO("name:%s", req.name.c_str());
+    // 设置返回数据
+    res.result = "OK";
+    // 订阅topic,注册回调函数show_data
+    ros::Subscriber client_topic_pub =
+            n.subscribe<std_msgs::Header>("head",10,show_data);
+}
+int main(int argc, char **argv) {
+    // 初始化节点
+    ros::init(argc, argv, "server_node");
+    // 设置节点句柄
+    ros::NodeHandle n;
+    // 创建名为"/show_name",的server,注册回调函数callback
+    ros::ServiceServer  name_server =
+            n.advertiseService("show_name", callback);
+    // 等待循环回调函数
+    ros::spin();
+    return 0;
+}
+/*
+void timeInfoCallback(const client::Time::ConstPtr& time) {
+    // 将接收到的消息打印出来
+    ROS_INFO("Topic_name:%s   time:%lu",
+             time->name.c_str(), time->sec);
+}
+bool nameCallback(client::show::Request &req,
+                  client::show::Response &res) {
+    res.rep = "client_topic";
+    ROS_INFO("%s", res.rep.c_str());
+    return true;
+}
+
+int main(int argc, char **argv) {
+    // 初始化ROS节点
+    ros::init(argc, argv, "server");
+    // 创建节点句柄
+    ros::NodeHandle n;
+    // 创建一个Subscriber，订阅名为client_topic的topic，注册回调函数timeInfoCallback
+    ros::Subscriber client_topic_sub = n.subscribe("/client_topic", 10,
+                                                   timeInfoCallback);
+    ros::ServiceServer service = n.advertiseService("show_name", nameCallback);
+    ros::spin();
+    return 0;
+}*/
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Could not find a newline character at the end of the file.  [whitespace/ending_newline] [5]